.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,233 @@
+name: build-and-test
+
+on: [push, pull_request]
+
+concurrency:
+  group: build-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  windows-test:
+    name: windows-test
+    runs-on: windows-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.2
+
+      - name: Install golang dependency
+        run: |
+          $ErrorActionPreference = 'Stop'
+          $env:GOPATH = "${env:USERPROFILE}\go"
+          go env -w CGO_ENABLED=0
+          go install github.com/ory/go-acc@v0.2.6
+
+      - name: Unit tests with coverage
+        run: |
+          $ErrorActionPreference = 'Stop'
+          $env:GOPATH = "${env:USERPROFILE}\go"
+          $env:PATH = "$env:PATH;${env:GOPATH}\bin"
+          go-acc ./...
+
+  installer-script-test:
+    name: installer-script-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PACKAGE_TYPE: [ "deb", "rpm" ]
+    env:
+      PYTHON_VERSION: 3.8.5
+      PIP_VERSION: 20.2.4
+      REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+      RESULT_PATH: "~/testresults"
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Check for relevant changes
```

Comment:
Can we try https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths instead?

Reply:
If we follow the way that you have suggested then we need to create different workflow files for each job.
Can we do that?

Reply:
Yes.  These jobs should be able to run independently and only when there are relevant changes.

Reply:
Changes made.